src/main/java/io/openliberty/tools/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -507,6 +510,39 @@ public void installFeatures(boolean isAcceptLicense, List<String> featuresToInst
                     throw new PluginExecutionException("Could not close resources after installing features.", e);
                 }
             }
+            restoreCacheInURLClassLoader();
+        }
+    }
+
+    // Attempt to disable connection caching in the URLClassLoader so that the jar files will
+    // all close when we close the class loader. Use reflection because this is not supported
+    // in Java 8. Save the current value to restore it later for performance reasons.
+    private synchronized void disableCacheInURLClassLoader() {
+        try {
+            if (saveURLCacheStatus == null) {
+                Method getDefaultCaching = java.net.URLConnection.class.getMethod("getDefaultUseCaches", String.class);
+                saveURLCacheStatus = Boolean.valueOf((boolean) getDefaultCaching.invoke(null, "jar")); // null = static method
+                Method disableCaching = java.net.URLConnection.class.getMethod("setDefaultUseCaches", String.class, boolean.class);
+                disableCaching.invoke(null, "jar", false); // null = static method, false = do not cache
+            }
+        } catch (Exception e) {  // ignore the exceptions if this does not work. E.g. no method found in Java 8.
+            info(e.getClass().getName() + " trying to invoke java.net.URLConnection.setDefaultUseCaches(S,b) in disable");
+        }
+    }
+
+    // Attempt to restore the connection caching value in the URLClassLoader so that the jar files
+    // will be cached or not as previously set. Use reflection because this is not supported
+    // in Java 8.
+    private synchronized void restoreCacheInURLClassLoader() {
+        try {
+            if (saveURLCacheStatus != null) {
+                Method disableCaching = java.net.URLConnection.class.getMethod("setDefaultUseCaches", String.class, boolean.class);
+                disableCaching.invoke(null, "jar", saveURLCacheStatus.booleanValue()); // null = static method
+            }
+        } catch (Exception e) {  // ignore the exceptions if this does not work. E.g. no method found in Java 8.
+            info(e.getClass().getName() + " trying to invoke java.net.URLConnection.setDefaultUseCaches(S,b) in restore");
```

Comment:
- Just catch `NoSuchMethodException` otherwise this will suppress any other possible exceptions.
- Log as debug instead of info, otherwise Java 8 users will always see this message.